src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+
+    }
+
+    private static class Node<T> {
+        private T item;
+        private MyLinkedList.Node<T> next;
+        private MyLinkedList.Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, MyLinkedList.Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> l = last;
```

Comment:
```suggestion
        Node<T> oldLast = last;
```
![image](https://user-images.githubusercontent.com/20326858/222116237-02c95e0f-19b7-432c-ac1f-a38e8162688d.png)